lib/ansible/plugins/action/reboot.py

```diff
@@ -240,9 +240,13 @@ def check_boot_time(self, distribution, previous_boot_time):
             try:
                 display.debug("{action}: setting connect_timeout to {value}".format(action=self._task.action, value=connect_timeout))
                 self._connection.set_option("connection_timeout", connect_timeout)
-                self._connection.reset()
-            except AttributeError:
-                display.warning("Connection plugin does not allow the connection timeout to be overridden")
+            except AnsibleError:
```

Comment:
Since `PluginBase.has_option` has existed since 2019, why not just check that first for `timeout` (since ssh is probably the more common usecase vs winrm/psrp), then `connection_timeout` if it's not there- set whichever one is supported (with no exception handler, actual set_option error should be fatal), or show the warning only if an explicit timeout was requested but the connection doesn't support it. Would simplify the exception handling logic on these a lot I think...

Reply:
this was made mostly to be backportable and not introduce new features, #82547 is where i'm toying with a 'nicer' way to deal with plugins not following a strict convention.